### ISSUE TITLE
MMR values not updating properly (linux)

### ISSUE THREAD

**Issue by mi12078** [AveYo/ShowMMR#2]
Values happen to update only after the game is (re)started.
Screenshot showing state prior to closing the game:
![image](https://github.com/AveYo/ShowMMR/assets/11778692/b4829598-aabe-476b-9f68-c003832b084a)

There is also suspicion that consecutive matches played in one session (meaning without restarting the game in-between) do not get their values updated at all, state after restarting the game:
![image](https://github.com/AveYo/ShowMMR/assets/11778692/ac9ef7c4-5d42-49f3-bbdd-34f78bfb53c4)

You can guess which games were played in one sitting by the times in screenshots above.

Installed the mod by copying over `game/dota_mods/pak01_dir.vpk` (did not exist before) and copied over `game/dota/gameinfo_branchspecific.gi` as well (kept the old file with a different name).

Running Dota/Steam client on Linux (Arch).

EDIT: just noticed commit [d5b57ca](https://github.com/AveYo/ShowMMR/commit/d5b57caaecb97e94bf690e49df76143dcd3f84fe) which most likely fixes the issue.

**Comment by AveYo** [comment]
Thank you for your detailed report!  
I was aware of the caching issue and panorama vs vscript events not being in sync  
With v1.3 hopefully it's all fixed, as it forces a more dynamic refresh just once after each game!

**Comment by AveYo** [comment]
Now that your history is "properly" messed up, you probably need
to delete / clear / adjust the dota/cfg/user_keys_[steamid]_slot3.vcfg manually

or try the optional ShowMMR tool to generate the actual history from steam  
compiling it on linux is not much different, could use [.NET 7.0 sdk for linux](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) then in ShowMMR_tool folder:
`dotnet publish -c Release -r linux-x64 --self-contained false -p:TargetFramework=net7.0`
and run it via dotnet run or browse to executable & chmod etc. **I should provide a .sh script as well!**

**Comment by mi12078** [comment]
Can confirm that with the above-mentioned commit the MMR change gets registered immediately after the last match was finished.
Ran the above command and the generated executable afterwards, got my MMR data populated for previous matches.

One nitpick though, for some matches the data was not rendered:
![image](https://github.com/AveYo/ShowMMR/assets/11778692/29863bc0-308e-4ec6-ad08-f08263e528ee)
whereas in the file that was needed to be copied over the data seemed to be present (snippet below):
```
...
203848110 { date 1687000038   mmr  2654   outcome    26 }
203308217 { date 1686963923   mmr  2628   outcome    25 }
203283367 { date 1686961116   mmr  2603   outcome   -25 }

203261364 { date 1686958612   mmr  2628   outcome   -25 }
203229009 { date 1686955380   mmr  2653   outcome   -24 }

203183515 { date 1686951586   mmr  2677   outcome    25 }
203132623 { date 1686948142   mmr  2652   outcome   -24 }
203077121 { date 1686944953   mmr  2676   outcome    26 }
...
```

**Comment by AveYo** [comment]
Unfortunately only the displayed match date & time is available to use as a key, and there have always been issues with it under linux (there was a launch option to address it that I don't remember).

Can't easily access the match id or the original epoch timestamp, so it's calculated by the script.
I've been searching for a more reliable way to match those.

Your input is quite valuable!
Maybe I can add another epoch calculation variant for such outliers.

**Comment by AveYo** [comment]
I think I have addressed it with v1.4
Thank you!

**Comment by mi12078** [comment]
Looks good on my end, thanks for the fix!
